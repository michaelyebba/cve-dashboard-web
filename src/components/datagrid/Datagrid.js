import React from 'react'
import CveRecord from './Cverecord'

let Datagrid = (props) => {

    return (

        <section id="report-section">
          <div class="container">
            <div class="report-title">
              <p class="h3">{props.title} <small class="muted">Common Vulnerabilities and Exposures (CVE)</small></p>
            </div>

            <div class="cve-table-container">
                <table class="table table-striped">
                  <thead class="thead-light">
                    <th scope="col">ID</th>
                    <th scope="col">SEVERITY</th>
                    <th scope="col">CATEGORY</th>
                    <th scope="col">PRODUCT(S)</th>
                    <th scope="col">DESCRIPTION</th>
                    <th scope="col">DATE</th>
                  </thead>
                  <tbody>
                    {props.cves.map((e,i) => {
                        return (
                            <CveRecord key={i} />
                        )
                    })}                          
                  </tbody>
                </table>
            </div>
          </div>
        </section>
    )
}

export default Datagrid;