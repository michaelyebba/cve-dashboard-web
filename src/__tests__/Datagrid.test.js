import React from 'react';
import {HashRouter} from 'react-router-dom'
import renderer from 'react-test-renderer';
import Datagrid from '../components/datagrid/Datagrid';

it('renders correctly', () => {

  let cves = [1,2,3,4]

  const tree = renderer
    .create(
        <HashRouter>
            <Datagrid cves={cves}/>
        </HashRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});