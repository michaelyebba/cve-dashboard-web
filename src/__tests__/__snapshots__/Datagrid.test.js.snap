// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  id="report-section"
>
  <div
    className="container"
  >
    <div
      className="report-title"
    >
      <p
        className="h3"
      >
         
        <small
          className="muted"
        >
          Common Vulnerabilities and Exposures (CVE)
        </small>
      </p>
    </div>
    <div
      className="cve-table-container"
    >
      <table
        className="table table-striped table-sm"
      >
        <thead
          className="thead-light"
        >
          <tr>
            <th
              scope="col"
            >
              ID
            </th>
            <th
              scope="col"
            >
              SEVERITY
            </th>
            <th
              scope="col"
            >
              CATEGORY
            </th>
            <th
              scope="col"
            >
              PRODUCT(S)
            </th>
            <th
              scope="col"
            >
              DATE
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              <a
                href="#/cve/undefined"
                onClick={[Function]}
              />
            </td>
            <td />
            <td />
            <td />
            <td />
          </tr>
          <tr>
            <td>
              <a
                href="#/cve/undefined"
                onClick={[Function]}
              />
            </td>
            <td />
            <td />
            <td />
            <td />
          </tr>
          <tr>
            <td>
              <a
                href="#/cve/undefined"
                onClick={[Function]}
              />
            </td>
            <td />
            <td />
            <td />
            <td />
          </tr>
          <tr>
            <td>
              <a
                href="#/cve/undefined"
                onClick={[Function]}
              />
            </td>
            <td />
            <td />
            <td />
            <td />
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</section>
`;
